# Top of /r/femalefashionadvice WAYWT

A script to get all the top comments in the WAYWT of a given month and format them for a post. This was formerly done manually by /u/thecurvynerd (thank you!). Example post: https://www.reddit.com/r/femalefashionadvice/comments/61241t/top_of_waywt_february_2017/

This script does NOT create the imgur album linked to in the title.

The example post, for reference. Jupyter notebook does not render `^` as superscripts.

## [Top of WAYWT: FEBRUARY 2017](http://imgur.com/a/t0468)

> 1\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8oszc/) ^(by */u/Wasmand*) ^**(+162)**  
2\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5vjhx0/waywt_february_22_2017/de2j8k6/) ^(by */u/Bloopity_bloop_bloop*) ^**(+141)**  
3\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5u89ok/waywt_february_15_2017/dds52c9/) ^(by */u/sohungy*) ^**(+130)**  
4\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5s847m/waywt_february_05_2017/ddd7aw1/) ^(by */u/smolherring*) ^**(+121)**  
5\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5rgaam/waywt_february_01_2017/dd7dgp6/) ^(by */u/shercocked*) ^**(+119)**  
6\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8q9sw/) ^(by */u/sohungy*) ^**(+109)**  
7\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5tm3q4/waywt_february_12_2017/ddniypv/) ^(by */u/BlueSpiritBlues*) ^**(+104)**  
8\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5rgaam/waywt_february_01_2017/dd72crh/) ^(by */u/ayiyi*) ^**(+103)**  
9\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5stiil/waywt_february_08_2017/ddhoahf/) ^(by */u/collapsedgovernment*) ^**(+101)**  
10\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5uyvqj/waywt_february_19_2017/ddy47x4/) ^(by */u/collapsedgovernment*) ^**(+100)**  
11\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5u89ok/waywt_february_15_2017/ddships/) ^(by */u/shercocked*) ^**(+99)**  
12\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5tm3q4/waywt_february_12_2017/ddnksu0/) ^(by */u/somethings_nothings*) ^**(+95)**  
13\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5vjhx0/waywt_february_22_2017/de2np2z/) ^(by */u/ayiyi*) ^**(+92)**  
14\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8quxg/) ^(by */u/SusannaCarmichael*) ^**(+92)**  
15\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5tm3q4/waywt_february_12_2017/ddnkz7v/) ^(by */u/HAPiper*) ^**(+91)**  
16\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5s847m/waywt_february_05_2017/ddd488f/) ^(by */u/BlueSpiritBlues*) ^**(+89)**  
17\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5stiil/waywt_february_08_2017/ddhp9gw/) ^(by */u/MonocleLewinsky*) ^**(+86)**  
18\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5u89ok/waywt_february_15_2017/ddsdf5o/) ^(by */u/saphydoodle*) ^**(+86)**  
19\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5stiil/waywt_february_08_2017/ddhq1oh/) ^(by */u/sohungy*) ^**(+83)**  
20\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5stiil/waywt_february_08_2017/ddhyjzl/) ^(by */u/skyllanyx*) ^**(+82)**  
21\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5uyvqj/waywt_february_19_2017/ddxw5ba/) ^(by */u/ceruleanmuse*) ^**(+80)**  
22\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5rgaam/waywt_february_01_2017/dd6yr0e/) ^(by */u/villonious*) ^**(+79)**  
23\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5u89ok/waywt_february_15_2017/dds54vw/) ^(by */u/BlueSpiritBlues*) ^**(+77)**  
24\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8mjjn/) ^(by */u/muzzyhoo92*) ^**(+75)**  
25\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5rgaam/waywt_february_01_2017/dd73qqj/) ^(by */u/earth_rises*) ^**(+72)**  
26\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5u89ok/waywt_february_15_2017/ddshqzt/) ^(by */u/imogenimpulse*) ^**(+72)**  
27\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5uyvqj/waywt_february_19_2017/ddxwku5/) ^(by */u/startupwoes*) ^**(+72)**  
28\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de93lv0/) ^(by */u/BlueSpiritBlues*) ^**(+71)**  
29\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8o2fb/) ^(by */u/villonious*) ^**(+69)**  
30\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5uyvqj/waywt_february_19_2017/ddy0dxj/) ^(by */u/Scrap-Icon*) ^**(+68)**  
31\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5tm3q4/waywt_february_12_2017/ddnrafw/) ^(by */u/villonious*) ^**(+67)**  
32\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5vjhx0/waywt_february_22_2017/de2o3h5/) ^(by */u/BlueSpiritBlues*) ^**(+67)**  
33\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5s847m/waywt_february_05_2017/ddd2ivh/) ^(by */u/aftertheswimmingpool*) ^**(+65)**  
34\. [Post](https://www.reddit.com/r/femalefashionadvice/comments/5vjhx0/waywt_february_22_2017/de2mlpz/) ^(by */u/averagefruit*) ^**(+65)**  


## Usage

### Knowledge Prerequisites

It will be a lot less painful to use this if you have experience using jupyter notebook. This section expects you know how to download this `ipynb` file (or the whole repository), install dependencies (manually, using a virtualenv, or using conda env) and set environment variables.

### Dependencies

This script requires the following to be installed on your computer:

- [python](https://www.python.org/downloads/) 3.5 (probably any python 3.x would work, but I used 3.5)
- [jupyter notebook](http://jupyter.org/install.html)
- [PRAW](https://praw.readthedocs.io/en/latest/index.html) (the Python Reddit API Wrapper).

There is a `environment.yml` file if you want to create a conda env (`conda env create -f environment.yml`) with jupyter and python 3.5. Conda does not have `praw` as of August 2017, so you will need to `pip install` that manually.

### Environment Variables

The reddit API requires you to register your app to be able to use it. You can get an app ID and app secret by following the steps [here](https://github.com/reddit/reddit/wiki/OAuth2-Quick-Start-Example#first-steps). Select 'script' when it asks what type of app it will be.

Save the app ID, app secret, and your reddit username as [environment variables](https://en.wikipedia.org/wiki/Environment_variable#Assignment). How you set these depends on what OS you're using.

```
export REDDIT_CLIENT_ID=your_id
export REDDIT_CLIENT_SECRET=your_secret
export REDDIT_USER=your_username
```

### Run the script

In the cell that says

```
# read-only subreddit instance
ffa = reddit.subreddit('femalefashionadvice')
# 1-indexed months. e.g. 2 would be February
MONTH = 2
YEAR = 2017
# How many comments you want in the 'Top of' list.
LIMIT = 34
```

Set `MONTH`, `YEAR`, and `LIMIT` to the values you want.

Unfortunately this script still requires manual intervention to filter out WAYWT posts within the month time range that are not, in fact, WAYWT posts. You can select the cell *under* the one that says `print_titles(month_submissions_with_waywt)`, and then from the notebook toolbar select `Cell` > `Run all above` to run all the cells up to that point. It should print a list of the submissions that contain 'WAYWT' in the title and were posted in the month you set. Here you can see that you would want to filter out things like previous "top of" posts. 

You can then change the variable `not_waywt = [3, 4, 7, 8, 10]` to have the numbers of the posts that you *don't* want to include. With that cell selected, you can then do `Run all below` to run the rest of the script.

You can paste the output of the `print(top_of_waywt_post)` cell in the bottom markdown cell to see what it would look like with formatting applied.

Note that deleted comments (if a comment author is `None`) are not included.

The code in this notebook is not the cleanest or most optimized because I am lazy, but hopefully it is still useful.

In [2]:
import calendar
import os

from datetime import datetime

import praw

In [3]:
def get_all_submissions_for_month(month, year):
    """
    Params
    ------
    month: int
    year: int
    """
    # https://stackoverflow.com/questions/42950/get-last-day-of-the-month-in-python
    last_day_of_month = calendar.monthrange(year, month)[1]

    unix_start_month = datetime.timestamp(datetime(year, month, 1))
    unix_end_month = datetime.timestamp(datetime(year, month, last_day_of_month, 23, 59, 59))
    
    return [post for post in ffa.submissions(start=unix_start_month, end=unix_end_month, extra_query=None)]


In [4]:
def get_all_submissions_for_year(year):
    unix_start_year = datetime.timestamp(datetime(year, 1, 1))
    unix_end_year = datetime.timestamp(datetime(year, 12, 31, 23, 59, 59))
    return [post for post in ffa.submissions(start=unix_start_year, end=unix_end_year, extra_query=None)]

In [19]:
def is_waywt(submission):
    return ('WAYWT' in submission.title) and \
           ('Announcement' not in submission.title) and \
           ('Top of' not in submission.title) and \
           ('x-post' not in submission.title)

In [6]:
def print_titles(submissions):
    # print for human checking
    if not submissions:
        print("No submissions to print!")
    else:
        for i,submission in enumerate(submissions):
            print('{0}. {1}'.format(i, submission.title))

In [7]:
def extract_comments(submission):
    # This will exclude the 'more comments' objects
    submission.comments.replace_more(limit=0)
    # Manually sort because setting submission.comment_sort to 'top' does not seem to actually sort by upvotes
    return [top_level_comment for top_level_comment in submission.comments if top_level_comment.author is not None]

In [8]:
def sort_comments_by_upvotes(comments):
    """comments is a list of pram Comments"""
    return sorted(comments, key=lambda x: x.ups, reverse=True)

In [9]:
def get_all_comments(submissions):
    all_comments = []
    for submission in submissions:
        all_comments += extract_comments(submission)
    return all_comments

In [37]:
def print_top_n_comments(limit, comments):
    """
    Params
    ------
    limit: int
    """
    info = "Author: /u/{0}\nUpvotes: {1}\nPermalink: {2}\n\n"
    for comment in comments[:limit]:
        print(info.format(comment.author, comment.ups, 'https://reddit.com'+comment.permalink))

In [11]:
# Instantiate your read-only reddit instance using PRAW
# You should have these set as environment variables
reddit = praw.Reddit(client_id=os.environ.get('REDDIT_CLIENT_ID'),
                     client_secret=os.environ.get('REDDIT_CLIENT_SECRET'),
                     user_agent=os.environ.get('REDDIT_USER'))

In [22]:
# read-only subreddit instance
ffa = reddit.subreddit('femalefashionadvice')

In [25]:
# 1-indexed months. e.g. 2 would be February
MONTH = 2
YEAR = 2017
# How many comments you want in the 'Top of' list.
LIMIT = 100

In [23]:
# month_submissions = get_all_submissions_for_month(MONTH, YEAR)
year_submissions = get_all_submissions_for_year(2017)

In [24]:
# month_submissions_with_waywt = [submission for submission in month_submissions if is_waywt(submission)]
year_submissions_with_waywt = [submission for submission in year_submissions if is_waywt(submission)]

In [26]:
# print_titles(month_submissions_with_waywt)
print_titles(year_submissions_with_waywt)

0. WAYWT - December 31, 2017
1. WAYWT - December 27, 2017
2. WAYWT - December 24, 2017
3. WAYWT - December 20, 2017
4. WAYWT - December 17, 2017
5. WAYWT - December 13, 2017
6. WAYWT - December 10, 2017
7. WAYWT - December 06, 2017
8. WAYWT - December 03, 2017
9. WAYWT - November 29, 2017
10. WAYWT - November 26, 2017
11. WAYWT - November 22, 2017
12. WAYWT - November 19, 2017
13. WAYWT - November 15, 2017
14. WAYWT - November 12, 2017
15. WAYWT - November 08, 2017
16. WAYWT - November 05, 2017
17. Month Long WAYWT Submission Thread
18. WAYWT - November 01, 2017
19. WAYWT - Halloween Edition!
20. WAYWT - October 29, 2017
21. WAYWT - October 25, 2017
22. WAYWT - October 22, 2017
23. WAYWT - October 18, 2017
24. WAYWT - October 15, 2017
25. WAYWT - October 11, 2017
26. WAYWT - October 08, 2017
27. WAYWT - October 04, 2017
28. Month Long WAYWT starts today!
29. WAYWT - October 01, 2017
30. WAYWT - September 27, 2017
31. Month-Long WAYWT will begin on Oct 1
32. WAYWT - September 24, 2017
3

In [29]:
# You probably need to manually intervene here
# Note that you need to stick to the zero-indexing as printed above.
# not_waywt = [3, 4, 7, 8, 10]
not_waywt = [28, 31, 35, 78, 82, 88, 92, 94, 124]
waywts_2017 = []
# for i,submission in enumerate(month_submissions_with_waywt):
for i,submission in enumerate(year_submissions_with_waywt):
    if i not in not_waywt:
        waywts_2017.append(submission)

print_titles(waywts_2017)

0. WAYWT - December 31, 2017
1. WAYWT - December 27, 2017
2. WAYWT - December 24, 2017
3. WAYWT - December 20, 2017
4. WAYWT - December 17, 2017
5. WAYWT - December 13, 2017
6. WAYWT - December 10, 2017
7. WAYWT - December 06, 2017
8. WAYWT - December 03, 2017
9. WAYWT - November 29, 2017
10. WAYWT - November 26, 2017
11. WAYWT - November 22, 2017
12. WAYWT - November 19, 2017
13. WAYWT - November 15, 2017
14. WAYWT - November 12, 2017
15. WAYWT - November 08, 2017
16. WAYWT - November 05, 2017
17. Month Long WAYWT Submission Thread
18. WAYWT - November 01, 2017
19. WAYWT - Halloween Edition!
20. WAYWT - October 29, 2017
21. WAYWT - October 25, 2017
22. WAYWT - October 22, 2017
23. WAYWT - October 18, 2017
24. WAYWT - October 15, 2017
25. WAYWT - October 11, 2017
26. WAYWT - October 08, 2017
27. WAYWT - October 04, 2017
28. WAYWT - October 01, 2017
29. WAYWT - September 27, 2017
30. WAYWT - September 24, 2017
31. WAYWT - September 20, 2017
32. WAYWT - September 17, 2017
33. WAYWT - Sep

In [30]:
# this might take a couple of seconds
all_top_level_comments = get_all_comments(waywts_2017)

In [32]:
sorted_comments = sort_comments_by_upvotes(all_top_level_comments)

In [38]:
print_top_n_comments(20, sorted_comments)

Author: /u/nipplemonger
Upvotes: 512
Permalink: https://reddit.com/r/femalefashionadvice/comments/6lw8m4/theme_waywt_submission_thread_dress_like_a_villain/djx5i30/


Author: /u/charity1021
Upvotes: 314
Permalink: https://reddit.com/r/femalefashionadvice/comments/79w8zw/waywt_halloween_edition/dp5dxa3/


Author: /u/minka92
Upvotes: 200
Permalink: https://reddit.com/r/femalefashionadvice/comments/6ooraw/theme_waywt_submission_dated/dkj4wp9/


Author: /u/full_boyle
Upvotes: 200
Permalink: https://reddit.com/r/femalefashionadvice/comments/69s0je/waywt_may_07_2017/dh990an/


Author: /u/pandawww
Upvotes: 194
Permalink: https://reddit.com/r/femalefashionadvice/comments/79w8zw/waywt_halloween_edition/dp5bnyb/


Author: /u/Wasmand
Upvotes: 176
Permalink: https://reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8oszc/


Author: /u/recycledstardust
Upvotes: 173
Permalink: https://reddit.com/r/femalefashionadvice/comments/6vjyte/waywt_august_23_2017/dm0to3g/


Author: /u/

In [42]:
def create_post_text(comments, annual=False):
    if annual:
        title = "Top 100 WAYWT of 2017\n".format(YEAR)
    else:
        title = "## TOP OF WAYWT: {0} {1}\n".format(calendar.month_name[MONTH], YEAR)

    # See https://www.reddit.com/wiki/commenting
    table_header = "Rank | Comment | User | Upvotes\n" + \
                   "---- | ---- | ---- | -------\n"

    row = "{rank} | [{post_title}](https://reddit.com{permalink}) | /u/{user} | +{upvotes}\n"

    post = title + '\n' + table_header

    # You could speed this up by refactoring to keep the submission title
    # with each comment, or by just making the permalink text just say 'link'.
    for i, comment in enumerate(comments):
        post += row.format(
            rank=i+1,
            permalink=comment.permalink,
            user=comment.author,
            upvotes=comment.ups,
            post_title=reddit.submission(id=comment._extract_submission_id()).title
        )
    return post

In [43]:
# This will take a little while
top_of_waywt_post = create_post_text(sorted_comments[:LIMIT], annual=True)

In [44]:
print(top_of_waywt_post)

Top 100 WAYWT of 2017

Rank | Comment | User | Upvotes
---- | ---- | ---- | -------
1 | [Theme WAYWT Submission Thread: Dress Like a Villain](https://reddit.com/r/femalefashionadvice/comments/6lw8m4/theme_waywt_submission_thread_dress_like_a_villain/djx5i30/) | /u/nipplemonger | +512
2 | [WAYWT - Halloween Edition!](https://reddit.com/r/femalefashionadvice/comments/79w8zw/waywt_halloween_edition/dp5dxa3/) | /u/charity1021 | +314
3 | [Theme WAYWT Submission: Dated](https://reddit.com/r/femalefashionadvice/comments/6ooraw/theme_waywt_submission_dated/dkj4wp9/) | /u/minka92 | +200
4 | [WAYWT - May 07, 2017](https://reddit.com/r/femalefashionadvice/comments/69s0je/waywt_may_07_2017/dh990an/) | /u/full_boyle | +200
5 | [WAYWT - Halloween Edition!](https://reddit.com/r/femalefashionadvice/comments/79w8zw/waywt_halloween_edition/dp5bnyb/) | /u/pandawww | +194
6 | [WAYWT - February 26, 2017](https://reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8oszc/) | /u/Wasmand 

Below is the output of the above in markdown, for human-checking that it came out right.

The script-generated output does not match with the original post. I think it's because the original post omitted the February 12th WAYWT. The upvotes do not match due to a time difference in when the data was collected (at the time people could have gone and upvoted the comments after seeing them via the 'Top of' post). /u/Bloopity_bloop_bloop deleted their comment since the original 'Top of' post, and it did not get included here.

## TOP OF WAYWT: February 2017

Rank | Comment | User | Upvotes
---- | ---- | ---- | -------
1 | [WAYWT - February 26, 2017](https://reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8oszc) | /u/Wasmand | +177
2 | [WAYWT - February 15, 2017](https://reddit.com/r/femalefashionadvice/comments/5u89ok/waywt_february_15_2017/dds52c9) | /u/sohungy | +140
3 | [WAYWT - February 05, 2017](https://reddit.com/r/femalefashionadvice/comments/5s847m/waywt_february_05_2017/ddd7aw1) | /u/smolherring | +128
4 | [WAYWT - February 01, 2017](https://reddit.com/r/femalefashionadvice/comments/5rgaam/waywt_february_01_2017/dd7dgp6) | /u/shercocked | +119
5 | [WAYWT - February 12, 2017](https://reddit.com/r/femalefashionadvice/comments/5tm3q4/waywt_february_12_2017/ddniypv) | /u/BlueSpiritBlues | +108
6 | [WAYWT - February 26, 2017](https://reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8q9sw) | /u/sohungy | +106
7 | [WAYWT - February 01, 2017](https://reddit.com/r/femalefashionadvice/comments/5rgaam/waywt_february_01_2017/dd72crh) | /u/ayiyi | +102
8 | [WAYWT - February 15, 2017](https://reddit.com/r/femalefashionadvice/comments/5u89ok/waywt_february_15_2017/ddships) | /u/shercocked | +97
9 | [WAYWT - February 26, 2017](https://reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8quxg) | /u/SusannaCarmichael | +94
10 | [WAYWT - February 22, 2017](https://reddit.com/r/femalefashionadvice/comments/5vjhx0/waywt_february_22_2017/de2np2z) | /u/ayiyi | +94
11 | [WAYWT - February 12, 2017](https://reddit.com/r/femalefashionadvice/comments/5tm3q4/waywt_february_12_2017/ddnksu0) | /u/somethings_nothings | +91
12 | [WAYWT - February 15, 2017](https://reddit.com/r/femalefashionadvice/comments/5u89ok/waywt_february_15_2017/ddsdf5o) | /u/saphydoodle | +87
13 | [WAYWT - February 05, 2017](https://reddit.com/r/femalefashionadvice/comments/5s847m/waywt_february_05_2017/ddd488f) | /u/BlueSpiritBlues | +86
14 | [WAYWT - February 08, 2017](https://reddit.com/r/femalefashionadvice/comments/5stiil/waywt_february_08_2017/ddhp9gw) | /u/MonocleLewinsky | +85
15 | [WAYWT - February 08, 2017](https://reddit.com/r/femalefashionadvice/comments/5stiil/waywt_february_08_2017/ddhq1oh) | /u/sohungy | +82
16 | [WAYWT - February 19, 2017](https://reddit.com/r/femalefashionadvice/comments/5uyvqj/waywt_february_19_2017/ddxw5ba) | /u/ceruleanmuse | +79
17 | [WAYWT - February 01, 2017](https://reddit.com/r/femalefashionadvice/comments/5rgaam/waywt_february_01_2017/dd6yr0e) | /u/villonious | +77
18 | [WAYWT - February 26, 2017](https://reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8mjjn) | /u/muzzyhoo92 | +76
19 | [WAYWT - February 15, 2017](https://reddit.com/r/femalefashionadvice/comments/5u89ok/waywt_february_15_2017/dds54vw) | /u/BlueSpiritBlues | +76
20 | [WAYWT - February 26, 2017](https://reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de93lv0) | /u/BlueSpiritBlues | +73
21 | [WAYWT - February 15, 2017](https://reddit.com/r/femalefashionadvice/comments/5u89ok/waywt_february_15_2017/ddshqzt) | /u/imogenimpulse | +73
22 | [WAYWT - February 01, 2017](https://reddit.com/r/femalefashionadvice/comments/5rgaam/waywt_february_01_2017/dd73qqj) | /u/earth_rises | +71
23 | [WAYWT - February 19, 2017](https://reddit.com/r/femalefashionadvice/comments/5uyvqj/waywt_february_19_2017/ddy0dxj) | /u/Scrap-Icon | +68
24 | [WAYWT - February 22, 2017](https://reddit.com/r/femalefashionadvice/comments/5vjhx0/waywt_february_22_2017/de2o3h5) | /u/BlueSpiritBlues | +67
25 | [WAYWT - February 12, 2017](https://reddit.com/r/femalefashionadvice/comments/5tm3q4/waywt_february_12_2017/ddnrafw) | /u/villonious | +67
26 | [WAYWT - February 26, 2017](https://reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8o2fb) | /u/villonious | +66
27 | [WAYWT - February 22, 2017](https://reddit.com/r/femalefashionadvice/comments/5vjhx0/waywt_february_22_2017/de2mlpz) | /u/averagefruit | +65
28 | [WAYWT - February 05, 2017](https://reddit.com/r/femalefashionadvice/comments/5s847m/waywt_february_05_2017/ddd2ivh) | /u/aftertheswimmingpool | +64
29 | [WAYWT - February 08, 2017](https://reddit.com/r/femalefashionadvice/comments/5stiil/waywt_february_08_2017/ddhoq9q) | /u/earth_rises | +63
30 | [WAYWT - February 01, 2017](https://reddit.com/r/femalefashionadvice/comments/5rgaam/waywt_february_01_2017/dd70bkl) | /u/useafuckingcoaster | +63
31 | [WAYWT - February 22, 2017](https://reddit.com/r/femalefashionadvice/comments/5vjhx0/waywt_february_22_2017/de2j361) | /u/omiiii | +61
32 | [WAYWT - February 19, 2017](https://reddit.com/r/femalefashionadvice/comments/5uyvqj/waywt_february_19_2017/ddxz0s8) | /u/BlueSpiritBlues | +59
33 | [WAYWT - February 01, 2017](https://reddit.com/r/femalefashionadvice/comments/5rgaam/waywt_february_01_2017/dd6zgdz) | /u/-lyrota- | +59
34 | [WAYWT - February 26, 2017](https://reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8pwc3) | /u/somethings_nothings | +58

## Top of WAYWT 2017

Top top-level comments in the WAYWT threads (including themes and Halloween)! The top WAYWT in /r/femalefashionadvice of 2017 goes to .......... /u/nipplemonger ¯\\\_(ツ)_/¯

Thanks to everyone for participating, whether you submitted fits or gave CC. There were 3898 top-level comments from 974 users (482 who only made one top-level comment). 

[Album](https://imgur.com/a/FGgLA) of part of the list. I was planning to add all 100 but imgur seems to be crapping out on me, so here are 38 of the first 50 or so that I added before the upload weirdness started. Links to each comment are in the table below for perusing the rest. For posts linking to multiple images, I only included the first one.

Some day I may automate the album creation.... I generated this post using [this script](http://github.com/margaret/top-of-waywt) (jupyter notebook I bashed out which is not representative of best practices of anything). Also note that since people can go back and vote on comments less than six months old, the points will likely change after this is posted as people go back through the old stuff.

Notes (for items with superscripts)

[1] Comment was not an outfit post

[2] Photo was removed

Rank | Comment | User | Upvotes
---- | ---- | ---- | -------
1 | [Theme WAYWT Submission Thread: Dress Like a Villain](https://reddit.com/r/femalefashionadvice/comments/6lw8m4/theme_waywt_submission_thread_dress_like_a_villain/djx5i30/) | /u/nipplemonger | +512
2 | [WAYWT - Halloween Edition!](https://reddit.com/r/femalefashionadvice/comments/79w8zw/waywt_halloween_edition/dp5dxa3/) | /u/charity1021 | +314
3 | [Theme WAYWT Submission: Dated^(1)](https://reddit.com/r/femalefashionadvice/comments/6ooraw/theme_waywt_submission_dated/dkj4wp9/) | /u/minka92 | +200
4 | [WAYWT - May 07, 2017](https://reddit.com/r/femalefashionadvice/comments/69s0je/waywt_may_07_2017/dh990an/) | /u/full_boyle | +200
5 | [WAYWT - Halloween Edition!^(2)](https://reddit.com/r/femalefashionadvice/comments/79w8zw/waywt_halloween_edition/dp5bnyb/) | /u/pandawww | +194
6 | [WAYWT - February 26, 2017](https://reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8oszc/) | /u/Wasmand | +176
7 | [WAYWT - August 23, 2017](https://reddit.com/r/femalefashionadvice/comments/6vjyte/waywt_august_23_2017/dm0to3g/) | /u/recycledstardust | +173
8 | [Theme WAYWT Submission Thread: Something's Not Right...](https://reddit.com/r/femalefashionadvice/comments/6g9aol/theme_waywt_submission_thread_somethings_not_right/diox8iv/) | /u/hopia89 | +171
9 | [WAYWT - Halloween Edition!](https://reddit.com/r/femalefashionadvice/comments/79w8zw/waywt_halloween_edition/dp5cszc/) | /u/maker_take_you | +170
10 | [WAYWT - Halloween Edition!](https://reddit.com/r/femalefashionadvice/comments/79w8zw/waywt_halloween_edition/dp5azg2/) | /u/ruthannr94 | +168
11 | [WAYWT - March 05, 2017](https://reddit.com/r/femalefashionadvice/comments/5xng3g/waywt_march_05_2017/dejgeyj/) | /u/ceruleanmuse | +163
12 | [WAYWT - November 29, 2017](https://reddit.com/r/femalefashionadvice/comments/7gebpd/waywt_november_29_2017/dqiew91/) | /u/carmen0042 | +160
13 | [WAYWT - April 30, 2017](https://reddit.com/r/femalefashionadvice/comments/68g6jq/waywt_april_30_2017/dgyjapg/) | /u/full_boyle | +156
14 | [WAYWT - August 13, 2017](https://reddit.com/r/femalefashionadvice/comments/6tfwpv/waywt_august_13_2017/dlklu15/) | /u/recycledstardust | +155
15 | [WAYWT - Halloween Edition!](https://reddit.com/r/femalefashionadvice/comments/79w8zw/waywt_halloween_edition/dp5nz8p/) | /u/Cubone_bone | +154
16 | [WAYWT - January 18, 2017](https://reddit.com/r/femalefashionadvice/comments/5oq65a/waywt_january_18_2017/dcl9mmf/) | /u/muzzyhoo92 | +153
17 | [Theme WAYWT Submission Thread: Dress Like a Villain](https://reddit.com/r/femalefashionadvice/comments/6lw8m4/theme_waywt_submission_thread_dress_like_a_villain/djxaggy/) | /u/CplElise | +151
18 | [Theme WAYWT Submission Thread: Dress Like a Villain](https://reddit.com/r/femalefashionadvice/comments/6lw8m4/theme_waywt_submission_thread_dress_like_a_villain/djx8e9b/) | /u/nAVEEE | +150
19 | [WAYWT - July 12, 2017](https://reddit.com/r/femalefashionadvice/comments/6mux2b/waywt_july_12_2017/dk4nwqm/) | /u/madinmadame | +149
20 | [WAYWT - August 02, 2017](https://reddit.com/r/femalefashionadvice/comments/6r56pb/waywt_august_02_2017/dl2j27y/) | /u/ayiyi | +145
21 | [WAYWT - May 31, 2017](https://reddit.com/r/femalefashionadvice/comments/6efxsi/waywt_may_31_2017/dia0bf6/) | /u/ayiyi | +144
22 | [WAYWT - October 22, 2017](https://reddit.com/r/femalefashionadvice/comments/7811tv/waywt_october_22_2017/doq5fkm/) | /u/coffee_for_dinner | +143
23 | [WAYWT - Halloween Edition!](https://reddit.com/r/femalefashionadvice/comments/79w8zw/waywt_halloween_edition/dp598qu/) | /u/thebigmeowski | +141
24 | [WAYWT - May 07, 2017](https://reddit.com/r/femalefashionadvice/comments/69s0je/waywt_may_07_2017/dh904vg/) | /u/saphydoodle | +141
25 | [WAYWT - December 20, 2017](https://reddit.com/r/femalefashionadvice/comments/7l2b2b/waywt_december_20_2017/drj4wiu/) | /u/niftynonsense | +140
26 | [WAYWT - October 29, 2017](https://reddit.com/r/femalefashionadvice/comments/79hah1/waywt_october_29_2017/dp22aii/) | /u/BaboonStick | +139
27 | [Theme WAYWT Submission Thread: Dress Like a Villain](https://reddit.com/r/femalefashionadvice/comments/6lw8m4/theme_waywt_submission_thread_dress_like_a_villain/djxezcc/) | /u/somethings_nothings | +139
28 | [WAYWT - March 26, 2017](https://reddit.com/r/femalefashionadvice/comments/61m9z7/waywt_march_26_2017/dfflpmn/) | /u/niftynonsense | +139
29 | [WAYWT - February 15, 2017](https://reddit.com/r/femalefashionadvice/comments/5u89ok/waywt_february_15_2017/dds52c9/) | /u/sohungy | +139
30 | [WAYWT - Halloween Edition!](https://reddit.com/r/femalefashionadvice/comments/79w8zw/waywt_halloween_edition/dp5gq30/) | /u/missgracie | +137
31 | [WAYWT - November 19, 2017](https://reddit.com/r/femalefashionadvice/comments/7e1fhr/waywt_november_19_2017/dq1vdy3/) | /u/niftynonsense | +136
32 | [WAYWT - March 19, 2017](https://reddit.com/r/femalefashionadvice/comments/60aqzk/waywt_march_19_2017/df56153/) | /u/Wasmand | +135
33 | [WAYWT - December 17, 2017](https://reddit.com/r/femalefashionadvice/comments/7kej8u/waywt_december_17_2017/drdradd/) | /u/0001909 | +134
34 | [WAYWT - December 06, 2017](https://reddit.com/r/femalefashionadvice/comments/7hz02m/waywt_december_06_2017/dquucoc/) | /u/carmen0042 | +134
35 | [WAYWT - June 21, 2017](https://reddit.com/r/femalefashionadvice/comments/6imsb7/waywt_june_21_2017/dj7epd0/) | /u/charity1021 | +133
36 | [Theme WAYWT Submission Thread: Dress Like Your Mom Wishes You Would](https://reddit.com/r/femalefashionadvice/comments/6as7fz/theme_waywt_submission_thread_dress_like_your_mom/dhhcsey/) | /u/ayiyi | +133
37 | [WAYWT - October 22, 2017](https://reddit.com/r/femalefashionadvice/comments/7811tv/waywt_october_22_2017/doq6yq1/) | /u/Deiji- | +132
38 | [WAYWT - June 14, 2017](https://reddit.com/r/femalefashionadvice/comments/6h8gqb/waywt_june_14_2017/diwmzaa/) | /u/Berryjammy | +132
39 | [WAYWT - December 17, 2017](https://reddit.com/r/femalefashionadvice/comments/7kej8u/waywt_december_17_2017/drdnv15/) | /u/gysss | +131
40 | [WAYWT - December 10, 2017](https://reddit.com/r/femalefashionadvice/comments/7iupvi/waywt_december_10_2017/dr1hiom/) | /u/0001909 | +131
41 | [WAYWT - July 23, 2017](https://reddit.com/r/femalefashionadvice/comments/6p22xw/waywt_july_23_2017/dkm2dql/) | /u/ayiyi | +131
42 | [WAYWT - July 09, 2017](https://reddit.com/r/femalefashionadvice/comments/6m87z8/waywt_july_09_2017/djzzc91/) | /u/full_boyle | +131
43 | [WAYWT - April 26, 2017](https://reddit.com/r/femalefashionadvice/comments/67p31j/waywt_april_26_2017/dgs5pv3/) | /u/babychickyellow | +131
44 | [Theme WAYWT Submission Thread: Art Period Inspired](https://reddit.com/r/femalefashionadvice/comments/6dhuaw/theme_waywt_submission_thread_art_period_inspired/di2qdz4/) | /u/nipplemonger | +130
45 | [WAYWT - September 06, 2017](https://reddit.com/r/femalefashionadvice/comments/6ygjqb/waywt_september_06_2017/dmnh34q/) | /u/cadmiumred | +128
46 | [WAYWT - February 05, 2017](https://reddit.com/r/femalefashionadvice/comments/5s847m/waywt_february_05_2017/ddd7aw1/) | /u/smolherring | +128
47 | [Theme WAYWT Submission: Dated^(1)](https://reddit.com/r/femalefashionadvice/comments/6ooraw/theme_waywt_submission_dated/dkj1kfr/) | /u/generalusername35 | +127
48 | [WAYWT - April 19, 2017](https://reddit.com/r/femalefashionadvice/comments/66b4q7/waywt_april_19_2017/dgh3mem/) | /u/ayiyi | +127
49 | [WAYWT - July 16, 2017](https://reddit.com/r/femalefashionadvice/comments/6nmxr1/waywt_july_16_2017/dkaok9t/) | /u/BaboonStick | +126
50 | [WAYWT - Halloween Edition!](https://reddit.com/r/femalefashionadvice/comments/79w8zw/waywt_halloween_edition/dp5nvd4/) | /u/pollydog | +125
51 | [WAYWT - January 29, 2017](https://reddit.com/r/femalefashionadvice/comments/5queyw/waywt_january_29_2017/dd2ap7j/) | /u/saphydoodle | +125
52 | [WAYWT - November 26, 2017](https://reddit.com/r/femalefashionadvice/comments/7fnqta/waywt_november_26_2017/dqd3543/) | /u/coffee_for_dinner | +122
53 | [Month Long WAYWT Submission Thread](https://reddit.com/r/femalefashionadvice/comments/7ak9q8/month_long_waywt_submission_thread/dpanxrq/) | /u/full_boyle | +121
54 | [WAYWT - July 19, 2017](https://reddit.com/r/femalefashionadvice/comments/6o9gk9/waywt_july_19_2017/dkfol2x/) | /u/emaleau | +121
55 | [WAYWT - April 05, 2017](https://reddit.com/r/femalefashionadvice/comments/63mgd1/waywt_april_05_2017/dfvb11s/) | /u/stephamelon | +121
56 | [WAYWT - June 07, 2017](https://reddit.com/r/femalefashionadvice/comments/6fudwa/waywt_june_07_2017/dil4nc7/) | /u/shercocked | +120
57 | [WAYWT - April 16, 2017](https://reddit.com/r/femalefashionadvice/comments/65ptu4/waywt_april_16_2017/dgca3ps/) | /u/saphydoodle | +120
58 | [WAYWT - January 22, 2017](https://reddit.com/r/femalefashionadvice/comments/5phooz/waywt_january_22_2017/dcrdygw/) | /u/xiaojyao | +120
59 | [WAYWT - December 03, 2017](https://reddit.com/r/femalefashionadvice/comments/7haezh/waywt_december_03_2017/dqplqtg/) | /u/charts_and_facts | +119
60 | [WAYWT - February 01, 2017](https://reddit.com/r/femalefashionadvice/comments/5rgaam/waywt_february_01_2017/dd7dgp6/) | /u/shercocked | +119
61 | [WAYWT - March 29, 2017](https://reddit.com/r/femalefashionadvice/comments/627j7s/waywt_march_29_2017/dfkbhtl/) | /u/ayiyi | +118
62 | [Theme WAYWT Submission Thread: Dress Like Your Mom Wishes You Would](https://reddit.com/r/femalefashionadvice/comments/6as7fz/theme_waywt_submission_thread_dress_like_your_mom/dhhbf0w/) | /u/Metcarfre | +117
63 | [WAYWT - April 12, 2017](https://reddit.com/r/femalefashionadvice/comments/64z2hk/waywt_april_12_2017/dg64tqc/) | /u/ayiyi | +117
64 | [WAYWT - November 26, 2017](https://reddit.com/r/femalefashionadvice/comments/7fnqta/waywt_november_26_2017/dqd2ot5/) | /u/0001909 | +116
65 | [WAYWT - Halloween Edition!](https://reddit.com/r/femalefashionadvice/comments/79w8zw/waywt_halloween_edition/dp6byav/) | /u/sewing_magic | +116
66 | [WAYWT - June 04, 2017](https://reddit.com/r/femalefashionadvice/comments/6f8395/waywt_june_04_2017/dig4ttb/) | /u/firstthingmonday | +116
67 | [Theme WAYWT Submission Thread: Art Period Inspired](https://reddit.com/r/femalefashionadvice/comments/6dhuaw/theme_waywt_submission_thread_art_period_inspired/di2rkgv/) | /u/zoxe | +116
68 | [WAYWT - November 08, 2017](https://reddit.com/r/femalefashionadvice/comments/7bm9d1/waywt_november_08_2017/dpj1d3c/) | /u/0001909 | +115
69 | [Theme WAYWT Submission Thread: Art Period Inspired](https://reddit.com/r/femalefashionadvice/comments/6dhuaw/theme_waywt_submission_thread_art_period_inspired/di2qlh3/) | /u/exmechanistic | +115
70 | [WAYWT - March 12, 2017](https://reddit.com/r/femalefashionadvice/comments/5yzbvg/waywt_march_12_2017/deu7vlo/) | /u/saphydoodle | +115
71 | [WAYWT - Halloween Edition!](https://reddit.com/r/femalefashionadvice/comments/79w8zw/waywt_halloween_edition/dp58gl0/) | /u/guineaparade | +114
72 | [WAYWT - March 26, 2017](https://reddit.com/r/femalefashionadvice/comments/61m9z7/waywt_march_26_2017/dffkbn5/) | /u/SusannaCarmichael | +114
73 | [WAYWT - March 05, 2017](https://reddit.com/r/femalefashionadvice/comments/5xng3g/waywt_march_05_2017/dejetqo/) | /u/muzzyhoo92 | +114
74 | [WAYWT - March 22, 2017](https://reddit.com/r/femalefashionadvice/comments/60vi51/waywt_march_22_2017/df9kjdt/) | /u/shercocked | +113
75 | [WAYWT - November 01, 2017](https://reddit.com/r/femalefashionadvice/comments/7a4jld/waywt_november_01_2017/dp74kih/) | /u/bean-sprout | +112
76 | [WAYWT - July 30, 2017](https://reddit.com/r/femalefashionadvice/comments/6qi4u7/waywt_july_30_2017/dkxu8f0/) | /u/sewing_magic | +112
77 | [WAYWT - July 19, 2017](https://reddit.com/r/femalefashionadvice/comments/6o9gk9/waywt_july_19_2017/dkfyr5x/) | /u/PM_ME_ITTYBITTYKITTY | +112
78 | [Theme WAYWT Submission Thread: Dress Like a Villain](https://reddit.com/r/femalefashionadvice/comments/6lw8m4/theme_waywt_submission_thread_dress_like_a_villain/djx68cy/) | /u/coffee_for_dinner | +111
79 | [WAYWT - April 16, 2017](https://reddit.com/r/femalefashionadvice/comments/65ptu4/waywt_april_16_2017/dgccac5/) | /u/goldiehawn | +111
80 | [WAYWT - January 01, 2017](https://reddit.com/r/femalefashionadvice/comments/5lfogz/waywt_january_01_2017/dbva9nc/) | /u/Straightgrizzly22 | +111
81 | [WAYWT - August 20, 2017](https://reddit.com/r/femalefashionadvice/comments/6uwiut/waywt_august_20_2017/dlwadct/) | /u/wanderedoff | +110
82 | [WAYWT - July 30, 2017](https://reddit.com/r/femalefashionadvice/comments/6qi4u7/waywt_july_30_2017/dkxkcr8/) | /u/Straightgrizzly22 | +110
83 | [WAYWT - January 01, 2017](https://reddit.com/r/femalefashionadvice/comments/5lfogz/waywt_january_01_2017/dbvbvwl/) | /u/abexaem | +110
84 | [WAYWT - December 31, 2017](https://reddit.com/r/femalefashionadvice/comments/7n981z/waywt_december_31_2017/ds00fyp/) | /u/coffee_for_dinner | +109
85 | [WAYWT - April 30, 2017](https://reddit.com/r/femalefashionadvice/comments/68g6jq/waywt_april_30_2017/dgy7gdo/) | /u/stephamelon | +108
86 | [WAYWT - March 05, 2017](https://reddit.com/r/femalefashionadvice/comments/5xng3g/waywt_march_05_2017/dejpq5h/) | /u/cikayelle | +108
87 | [WAYWT - February 12, 2017](https://reddit.com/r/femalefashionadvice/comments/5tm3q4/waywt_february_12_2017/ddniypv/) | /u/BlueSpiritBlues | +108
88 | [WAYWT - August 09, 2017](https://reddit.com/r/femalefashionadvice/comments/6smgup/waywt_august_09_2017/dldx97t/) | /u/emaleau | +107
89 | [WAYWT - July 19, 2017](https://reddit.com/r/femalefashionadvice/comments/6o9gk9/waywt_july_19_2017/dkfpjar/) | /u/Lillyeth | +106
90 | [WAYWT - May 17, 2017](https://reddit.com/r/femalefashionadvice/comments/6bpmgl/waywt_may_17_2017/dhom2ir/) | /u/skyllanyx | +106
91 | [WAYWT - March 19, 2017](https://reddit.com/r/femalefashionadvice/comments/60aqzk/waywt_march_19_2017/df4vai9/) | /u/somethings_nothings | +106
92 | [WAYWT - March 15, 2017](https://reddit.com/r/femalefashionadvice/comments/5zk88l/waywt_march_15_2017/deyqcvk/) | /u/ayiyi | +106
93 | [WAYWT - October 22, 2017](https://reddit.com/r/femalefashionadvice/comments/7811tv/waywt_october_22_2017/doqgzrj/) | /u/full_boyle | +105
94 | [WAYWT - July 02, 2017](https://reddit.com/r/femalefashionadvice/comments/6ktzj8/waywt_july_02_2017/djosxbk/) | /u/salted_sky | +105
95 | [Theme WAYWT Submission Thread: All Grown Up](https://reddit.com/r/femalefashionadvice/comments/683jiy/theme_waywt_submission_thread_all_grown_up/dgvhq3p/) | /u/gnarlykitten | +105
96 | [WAYWT - March 26, 2017](https://reddit.com/r/femalefashionadvice/comments/61m9z7/waywt_march_26_2017/dffoly1/) | /u/BlueSpiritBlues | +105
97 | [WAYWT - February 26, 2017](https://reddit.com/r/femalefashionadvice/comments/5waqux/waywt_february_26_2017/de8q9sw/) | /u/sohungy | +105
98 | [WAYWT - May 24, 2017](https://reddit.com/r/femalefashionadvice/comments/6d3352/waywt_may_24_2017/dhzicjz/) | /u/Wasmand | +104
99 | [WAYWT - May 10, 2017](https://reddit.com/r/femalefashionadvice/comments/6adozz/waywt_may_10_2017/dhdrbqm/) | /u/super_cheeky | +104
100 | [WAYWT - March 29, 2017](https://reddit.com/r/femalefashionadvice/comments/627j7s/waywt_march_29_2017/dfkebuu/) | /u/DrNooooooo | +104

## More stats!

### Most posts

rank | user | total posts
---- | ---- | -----------
1 | /u/emaleau | 80
2 | /u/coffee_for_dinner | 74
3 | /u/bikinikills | 59
4 | /u/ruthannr94 | 56
5 | /u/xomandark | 49
6 | /u/heyyogagirl | 44
7 | /u/sarahaw13 | 42
8 | /u/xtinamariet | 42
9 | /u/LisaXavier | 42
10 | /u/kcwashere | 41
11 | /u/Deiji- | 39
12 | /u/Daniellewhatever | 39
13 | /u/respectablerag | 38
14 | /u/e___e | 38
15 | /u/LizzyLemonade | 38
16 | /u/exmechanistic | 37
17 | /u/villonious | 35
18 | /u/shercocked | 34
19 | /u/undercoverofbranches | 33
20 | /u/useafuckingcoaster | 32
21 | /u/thecurvynerd | 31
22 | /u/lizzybones | 28
23 | /u/BlueSpiritBlues | 27
24 | /u/Lillyeth | 25
25 | /u/ratioyouknow | 25
26 | /u/302149 | 24
27 | /u/niftynonsense | 24
28 | /u/robotgwen | 24
29 | /u/hopia89 | 24
30 | /u/asdfjklOHFUCKYOU | 24
31 | /u/cheesegrater_knees | 21
32 | /u/ModestMalka | 20
33 | /u/foxlie | 20
34 | /u/Hoosiergirl29 | 19
35 | /u/recycledstardust | 18
36 | /u/Gabwielle | 18
37 | /u/pygoscelis | 18
38 | /u/stephamelon | 18
39 | /u/somethings_nothings | 18
40 | /u/full_boyle | 18
41 | /u/railaway | 18
42 | /u/ediblesprysky | 18
43 | /u/twoblueshoes | 17
44 | /u/chaosintejas | 17
45 | /u/BaboonStick | 16
46 | /u/ticklytiger | 16
47 | /u/tamberra | 16
48 | /u/earth_rises | 16
49 | /u/tomlizzo | 16
50 | /u/lawschoolbimbo | 15
51 | /u/sheena2015 | 15
52 | /u/kinasekinase | 15
53 | /u/catherine_wheel | 15
54 | /u/cinnathymee | 15
55 | /u/eyeslikeashowroom | 15
56 | /u/salted_sky | 15
57 | /u/seriicis | 14
58 | /u/bean-sprout | 14
59 | /u/WiseacreSlugabed | 14
60 | /u/w0nd3rlust | 14
61 | /u/layneepup | 14
62 | /u/spacebarormarsbar | 13
63 | /u/memziejane | 13
64 | /u/caffeinatedlackey | 13
65 | /u/sklaundere | 13
66 | /u/super_cheeky | 13
67 | /u/fitzpleasr | 13
68 | /u/Rogue_Fibre | 13
69 | /u/CinnamonBunBun | 13
70 | /u/ayiyi | 13
71 | /u/ottaline_gambol | 12
72 | /u/2000srh | 12
73 | /u/muzzyhoo92 | 12
74 | /u/spookybuttwhispers | 12
75 | /u/domesticenginerd | 12
76 | /u/charts_and_facts | 11
77 | /u/tuliphaze | 11
78 | /u/totesk | 11
79 | /u/scooby_noob | 10
80 | /u/AryaTS | 10
81 | /u/saphydoodle | 10
82 | /u/0001909 | 10
83 | /u/Lipstickandpixiedust | 10
84 | /u/HelloSwede | 10
85 | /u/KodiakTheBear9 | 10
86 | /u/ceruleanmuse | 10
87 | /u/aliceismalice | 10
88 | /u/catsincaves | 10
89 | /u/red_raconteur | 9
90 | /u/bikehelmetfashion | 9
91 | /u/Fiberista | 9
92 | /u/cusekyi | 9
93 | /u/-beyonce- | 9
94 | /u/madinmadame | 9
95 | /u/KimchiChan | 9
96 | /u/queenbeluga | 9
97 | /u/faux-kwah | 9
98 | /u/redreplicant | 9
99 | /u/kenyan-girl | 9
100 | /u/wanderedoff | 9

### Most points

rank | user | total points | number of posts
---- | ---- | ------------ | ---------------
1 | /u/coffee_for_dinner | 3628 | (74 posts)
2 | /u/emaleau | 3607 | (80 posts)
3 | /u/shercocked | 2465 | (34 posts)
4 | /u/ruthannr94 | 2386 | (56 posts)
5 | /u/e___e | 2183 | (38 posts)
6 | /u/kcwashere | 1991 | (41 posts)
7 | /u/BlueSpiritBlues | 1904 | (27 posts)
8 | /u/sarahaw13 | 1785 | (42 posts)
9 | /u/full_boyle | 1726 | (18 posts)
10 | /u/villonious | 1707 | (35 posts)
11 | /u/xomandark | 1703 | (49 posts)
12 | /u/niftynonsense | 1625 | (24 posts)
13 | /u/Deiji- | 1596 | (39 posts)
14 | /u/ayiyi | 1448 | (13 posts)
15 | /u/recycledstardust | 1411 | (18 posts)
16 | /u/useafuckingcoaster | 1373 | (32 posts)
17 | /u/exmechanistic | 1349 | (37 posts)
18 | /u/bikinikills | 1315 | (59 posts)
19 | /u/LisaXavier | 1312 | (42 posts)
20 | /u/thecurvynerd | 1230 | (31 posts)
21 | /u/somethings_nothings | 1203 | (18 posts)
22 | /u/hopia89 | 1170 | (24 posts)
23 | /u/heyyogagirl | 1107 | (44 posts)
24 | /u/Lillyeth | 1079 | (25 posts)
25 | /u/respectablerag | 1059 | (38 posts)
26 | /u/lizzybones | 1032 | (28 posts)
27 | /u/saphydoodle | 982 | (10 posts)
28 | /u/Daniellewhatever | 978 | (39 posts)
29 | /u/foxlie | 977 | (20 posts)
30 | /u/0001909 | 965 | (10 posts)
31 | /u/bean-sprout | 939 | (14 posts)
32 | /u/ModestMalka | 891 | (20 posts)
33 | /u/BaboonStick | 886 | (16 posts)
34 | /u/muzzyhoo92 | 881 | (12 posts)
35 | /u/stephamelon | 848 | (18 posts)
36 | /u/Gabwielle | 842 | (18 posts)
37 | /u/ratioyouknow | 812 | (25 posts)
38 | /u/LizzyLemonade | 771 | (38 posts)
39 | /u/chaosintejas | 757 | (17 posts)
40 | /u/super_cheeky | 744 | (13 posts)
41 | /u/earth_rises | 718 | (16 posts)
42 | /u/salted_sky | 705 | (15 posts)
43 | /u/nipplemonger | 675 | (3 posts)
44 | /u/tomlizzo | 664 | (16 posts)
45 | /u/cheesegrater_knees | 640 | (21 posts)
46 | /u/ceruleanmuse | 639 | (10 posts)
47 | /u/xtinamariet | 618 | (42 posts)
48 | /u/SusannaCarmichael | 600 | (8 posts)
49 | /u/madinmadame | 595 | (9 posts)
50 | /u/babychickyellow | 594 | (8 posts)
51 | /u/charts_and_facts | 585 | (11 posts)
52 | /u/Hoosiergirl29 | 583 | (19 posts)
53 | /u/sohungy | 581 | (7 posts)
54 | /u/fitzpleasr | 561 | (13 posts)
55 | /u/Wasmand | 559 | (5 posts)
56 | /u/layneepup | 555 | (14 posts)
57 | /u/seriicis | 543 | (14 posts)
58 | /u/catherine_wheel | 538 | (15 posts)
59 | /u/faux-kwah | 531 | (9 posts)
60 | /u/eyeslikeashowroom | 523 | (15 posts)
61 | /u/tamberra | 521 | (16 posts)
62 | /u/Straightgrizzly22 | 520 | (7 posts)
63 | /u/goldiehawn | 515 | (6 posts)
64 | /u/pygoscelis | 489 | (18 posts)
65 | /u/charity1021 | 478 | (3 posts)
66 | /u/gombrick | 466 | (8 posts)
67 | /u/2000srh | 464 | (12 posts)
68 | /u/robotgwen | 464 | (24 posts)
69 | /u/sheena2015 | 443 | (15 posts)
70 | /u/wanderedoff | 440 | (9 posts)
71 | /u/KodiakTheBear9 | 437 | (10 posts)
72 | /u/corinnaps | 433 | (7 posts)
73 | /u/302149 | 433 | (24 posts)
74 | /u/ottaline_gambol | 419 | (12 posts)
75 | /u/tuliphaze | 416 | (11 posts)
76 | /u/Fiberista | 405 | (9 posts)
77 | /u/ticklytiger | 400 | (16 posts)
78 | /u/asdfjklOHFUCKYOU | 387 | (24 posts)
79 | /u/Scrap-Icon | 382 | (8 posts)
80 | /u/firstthingmonday | 374 | (8 posts)
81 | /u/miss-nebula | 371 | (6 posts)
82 | /u/cinnathymee | 368 | (15 posts)
83 | /u/caffeinatedlackey | 368 | (13 posts)
84 | /u/kinasekinase | 361 | (15 posts)
85 | /u/ediblesprysky | 355 | (18 posts)
86 | /u/-beyonce- | 351 | (9 posts)
87 | /u/w0nd3rlust | 350 | (14 posts)
88 | /u/KimchiChan | 349 | (9 posts)
89 | /u/PM_ME_ITTYBITTYKITTY | 335 | (4 posts)
90 | /u/undercoverofbranches | 329 | (33 posts)
91 | /u/carmen0042 | 326 | (4 posts)
92 | /u/imogenimpulse | 316 | (6 posts)
93 | /u/always-_-sleepy | 313 | (6 posts)
94 | /u/darkitecture | 313 | (6 posts)
95 | /u/Rogue_Fibre | 310 | (13 posts)
96 | /u/lawschoolbimbo | 302 | (15 posts)
97 | /u/hiilive | 298 | (5 posts)
98 | /u/ahrieru | 298 | (5 posts)
99 | /u/suckckit | 288 | (6 posts)
100 | /u/MoonlightBloom | 288 | (7 posts)

### Most points per post

rank | user | points per post | number of posts
---- | ---- | --------------- | ---------------
1 | /u/nipplemonger | 225.0 | (3 posts)
2 | /u/minka92 | 200.0 | (1 posts)
3 | /u/pandawww | 194.0 | (1 posts)
4 | /u/maker_take_you | 170.0 | (1 posts)
5 | /u/charity1021 | 159.33333333333334 | (3 posts)
6 | /u/Cubone_bone | 154.0 | (1 posts)
7 | /u/CplElise | 151.0 | (1 posts)
8 | /u/nAVEEE | 150.0 | (1 posts)
9 | /u/thebigmeowski | 141.0 | (1 posts)
10 | /u/missgracie | 137.0 | (1 posts)
11 | /u/generalusername35 | 127.0 | (1 posts)
12 | /u/pollydog | 125.0 | (1 posts)
13 | /u/Metcarfre | 117.0 | (1 posts)
14 | /u/zoxe | 116.0 | (1 posts)
15 | /u/sewing_magic | 114.0 | (2 posts)
16 | /u/Wasmand | 111.8 | (5 posts)
17 | /u/ayiyi | 111.38461538461539 | (13 posts)
18 | /u/gnarlykitten | 105.0 | (1 posts)
19 | /u/DrNooooooo | 104.0 | (1 posts)
20 | /u/pilnock | 100.0 | (1 posts)
21 | /u/xiaojyao | 100.0 | (2 posts)
22 | /u/skyllanyx | 99.5 | (2 posts)
23 | /u/saphydoodle | 98.2 | (10 posts)
24 | /u/orlando_furiosa | 97.0 | (1 posts)
25 | /u/0001909 | 96.5 | (10 posts)
26 | /u/nachtmere | 96.0 | (1 posts)
27 | /u/full_boyle | 95.88888888888889 | (18 posts)
28 | /u/LentilBurguer | 95.0 | (1 posts)
29 | /u/Bloopity_bloop_bloop | 94.0 | (1 posts)
30 | /u/nnutella | 91.0 | (1 posts)
31 | /u/theteenagegentleman | 90.0 | (1 posts)
32 | /u/cadmiumred | 87.0 | (2 posts)
33 | /u/goldiehawn | 85.83333333333333 | (6 posts)
34 | /u/PM_ME_ITTYBITTYKITTY | 83.75 | (4 posts)
35 | /u/sohungy | 83.0 | (7 posts)
36 | /u/social_potato | 83.0 | (1 posts)
37 | /u/carmen0042 | 81.5 | (4 posts)
38 | /u/chameshi_nampa | 81.0 | (1 posts)
39 | /u/maya1987_K | 81.0 | (3 posts)
40 | /u/cccoffeeegurl | 80.0 | (1 posts)
41 | /u/mga92 | 80.0 | (1 posts)
42 | /u/recycledstardust | 78.38888888888889 | (18 posts)
43 | /u/gal_gardner | 78.0 | (2 posts)
44 | /u/therealtompetty | 77.0 | (1 posts)
45 | /u/amber0517 | 76.0 | (1 posts)
46 | /u/viabrera | 76.0 | (1 posts)
47 | /u/SusannaCarmichael | 75.0 | (8 posts)
48 | /u/any_delirium | 75.0 | (1 posts)
49 | /u/smolherring | 74.33333333333333 | (3 posts)
50 | /u/Straightgrizzly22 | 74.28571428571429 | (7 posts)
51 | /u/babychickyellow | 74.25 | (8 posts)
52 | /u/Berryjammy | 73.5 | (2 posts)
53 | /u/muzzyhoo92 | 73.41666666666667 | (12 posts)
54 | /u/shercocked | 72.5 | (34 posts)
55 | /u/filletoxico | 72.0 | (2 posts)
56 | /u/ilovemrmiyagi | 72.0 | (1 posts)
57 | /u/rch-ie | 71.0 | (1 posts)
58 | /u/BlueSpiritBlues | 70.51851851851852 | (27 posts)
59 | /u/sudosussudio | 70.0 | (1 posts)
60 | /u/Sun_Sprout | 70.0 | (1 posts)
61 | /u/feminines | 69.5 | (2 posts)
62 | /u/clankton | 69.0 | (1 posts)
63 | /u/mycatispipeyq | 68.0 | (4 posts)
64 | /u/_ad_nauseam | 68.0 | (1 posts)
65 | /u/niftynonsense | 67.70833333333333 | (24 posts)
66 | /u/bean-sprout | 67.07142857142857 | (14 posts)
67 | /u/AVOCADOCUPCAKES | 67.0 | (4 posts)
68 | /u/somethings_nothings | 66.83333333333333 | (18 posts)
69 | /u/ohmamie | 66.5 | (2 posts)
70 | /u/madinmadame | 66.11111111111111 | (9 posts)
71 | /u/Olibrow14 | 66.0 | (1 posts)
72 | /u/casevintageshop | 66.0 | (1 posts)
73 | /u/thane311 | 65.0 | (1 posts)
74 | /u/jjcreskoff | 65.0 | (1 posts)
75 | /u/averagefruit | 64.0 | (1 posts)
76 | /u/ceruleanmuse | 63.9 | (10 posts)
77 | /u/WesterosiAssassin | 63.0 | (1 posts)
78 | /u/Kkal73 | 62.5 | (2 posts)
79 | /u/b_writes | 62.5 | (4 posts)
80 | /u/maydaytuesday | 62.0 | (1 posts)
81 | /u/corinnaps | 61.857142857142854 | (7 posts)
82 | /u/miss-nebula | 61.833333333333336 | (6 posts)
83 | /u/yesnomaybe-idontknow | 61.0 | (1 posts)
84 | /u/DigitalClutter | 60.0 | (1 posts)
85 | /u/hiilive | 59.6 | (5 posts)
86 | /u/ahrieru | 59.6 | (5 posts)
87 | /u/hand2handwombat | 59.5 | (4 posts)
88 | /u/faux-kwah | 59.0 | (9 posts)
89 | /u/gombrick | 58.25 | (8 posts)
90 | /u/691337 | 58.0 | (1 posts)
91 | /u/evrtgurnot | 58.0 | (2 posts)
92 | /u/e___e | 57.44736842105263 | (38 posts)
93 | /u/super_cheeky | 57.23076923076923 | (13 posts)
94 | /u/warpweftwatergate | 57.0 | (1 posts)
95 | /u/pamperedpunk | 57.0 | (1 posts)
96 | /u/oy_ | 56.666666666666664 | (3 posts)
97 | /u/papercranium | 56.5 | (2 posts)
98 | /u/EpicShitTime | 56.0 | (1 posts)
99 | /u/TopGagne | 56.0 | (1 posts)
100 | /u/BaboonStick | 55.375 | (16 posts)

In [111]:
from collections import defaultdict
users = defaultdict(lambda:{'points':0, 'count':0})
for comment in sorted_comments:
    user = str(comment.author)
    users[user]['points'] += comment.ups
    users[user]['count'] += 1

In [112]:
by_count = sorted(users, key=lambda x:users[x]['count'], reverse=True)
print('rank | user | total posts')
print('---- | ---- | -----------')
for i, user in enumerate(by_count[:100]):
    print("{} | /u/{} | {}".format(i+1, user, users[user]['count']))

rank | user | total posts
---- | ---- | -----------
1 | /u/emaleau | 80
2 | /u/coffee_for_dinner | 74
3 | /u/bikinikills | 59
4 | /u/ruthannr94 | 56
5 | /u/xomandark | 49
6 | /u/heyyogagirl | 44
7 | /u/sarahaw13 | 42
8 | /u/xtinamariet | 42
9 | /u/LisaXavier | 42
10 | /u/kcwashere | 41
11 | /u/Deiji- | 39
12 | /u/Daniellewhatever | 39
13 | /u/respectablerag | 38
14 | /u/e___e | 38
15 | /u/LizzyLemonade | 38
16 | /u/exmechanistic | 37
17 | /u/villonious | 35
18 | /u/shercocked | 34
19 | /u/undercoverofbranches | 33
20 | /u/useafuckingcoaster | 32
21 | /u/thecurvynerd | 31
22 | /u/lizzybones | 28
23 | /u/BlueSpiritBlues | 27
24 | /u/Lillyeth | 25
25 | /u/ratioyouknow | 25
26 | /u/302149 | 24
27 | /u/niftynonsense | 24
28 | /u/robotgwen | 24
29 | /u/hopia89 | 24
30 | /u/asdfjklOHFUCKYOU | 24
31 | /u/cheesegrater_knees | 21
32 | /u/ModestMalka | 20
33 | /u/foxlie | 20
34 | /u/Hoosiergirl29 | 19
35 | /u/recycledstardust | 18
36 | /u/Gabwielle | 18
37 | /u/pygoscelis | 18
38 | /u/stephamelon

rank | user | total posts
---- | ---- | -----------
1 | /u/emaleau | 80
2 | /u/coffee_for_dinner | 74
3 | /u/bikinikills | 59
4 | /u/ruthannr94 | 56
5 | /u/xomandark | 49
6 | /u/heyyogagirl | 44
7 | /u/sarahaw13 | 42
8 | /u/xtinamariet | 42
9 | /u/LisaXavier | 42
10 | /u/kcwashere | 41
11 | /u/Deiji- | 39
12 | /u/Daniellewhatever | 39
13 | /u/respectablerag | 38
14 | /u/e___e | 38
15 | /u/LizzyLemonade | 38
16 | /u/exmechanistic | 37
17 | /u/villonious | 35
18 | /u/shercocked | 34
19 | /u/undercoverofbranches | 33
20 | /u/useafuckingcoaster | 32
21 | /u/thecurvynerd | 31
22 | /u/lizzybones | 28
23 | /u/BlueSpiritBlues | 27
24 | /u/Lillyeth | 25
25 | /u/ratioyouknow | 25
26 | /u/302149 | 24
27 | /u/niftynonsense | 24
28 | /u/robotgwen | 24
29 | /u/hopia89 | 24
30 | /u/asdfjklOHFUCKYOU | 24
31 | /u/cheesegrater_knees | 21
32 | /u/ModestMalka | 20
33 | /u/foxlie | 20
34 | /u/Hoosiergirl29 | 19
35 | /u/recycledstardust | 18
36 | /u/Gabwielle | 18
37 | /u/pygoscelis | 18
38 | /u/stephamelon | 18
39 | /u/somethings_nothings | 18
40 | /u/full_boyle | 18
41 | /u/railaway | 18
42 | /u/ediblesprysky | 18
43 | /u/twoblueshoes | 17
44 | /u/chaosintejas | 17
45 | /u/BaboonStick | 16
46 | /u/ticklytiger | 16
47 | /u/tamberra | 16
48 | /u/earth_rises | 16
49 | /u/tomlizzo | 16
50 | /u/lawschoolbimbo | 15
51 | /u/sheena2015 | 15
52 | /u/kinasekinase | 15
53 | /u/catherine_wheel | 15
54 | /u/cinnathymee | 15
55 | /u/eyeslikeashowroom | 15
56 | /u/salted_sky | 15
57 | /u/seriicis | 14
58 | /u/bean-sprout | 14
59 | /u/WiseacreSlugabed | 14
60 | /u/w0nd3rlust | 14
61 | /u/layneepup | 14
62 | /u/spacebarormarsbar | 13
63 | /u/memziejane | 13
64 | /u/caffeinatedlackey | 13
65 | /u/sklaundere | 13
66 | /u/super_cheeky | 13
67 | /u/fitzpleasr | 13
68 | /u/Rogue_Fibre | 13
69 | /u/CinnamonBunBun | 13
70 | /u/ayiyi | 13
71 | /u/ottaline_gambol | 12
72 | /u/2000srh | 12
73 | /u/muzzyhoo92 | 12
74 | /u/spookybuttwhispers | 12
75 | /u/domesticenginerd | 12
76 | /u/charts_and_facts | 11
77 | /u/tuliphaze | 11
78 | /u/totesk | 11
79 | /u/scooby_noob | 10
80 | /u/AryaTS | 10
81 | /u/saphydoodle | 10
82 | /u/0001909 | 10
83 | /u/Lipstickandpixiedust | 10
84 | /u/HelloSwede | 10
85 | /u/KodiakTheBear9 | 10
86 | /u/ceruleanmuse | 10
87 | /u/aliceismalice | 10
88 | /u/catsincaves | 10
89 | /u/red_raconteur | 9
90 | /u/bikehelmetfashion | 9
91 | /u/Fiberista | 9
92 | /u/cusekyi | 9
93 | /u/-beyonce- | 9
94 | /u/madinmadame | 9
95 | /u/KimchiChan | 9
96 | /u/queenbeluga | 9
97 | /u/faux-kwah | 9
98 | /u/redreplicant | 9
99 | /u/kenyan-girl | 9
100 | /u/wanderedoff | 9


In [113]:
by_points = sorted(users, key=lambda x:users[x]['points'], reverse=True)
print('rank | user | total points | number of posts')
print('---- | ---- | ------------ | ---------------')
for i,user in enumerate(by_points[:100]):
    print("{} | /u/{} | {} | ({} posts)".format(i+1, user, users[user]['points'], users[user]['count']))

rank | user | total points | number of posts
---- | ---- | ------------ | ---------------
1 | /u/coffee_for_dinner | 3628 | (74 posts)
2 | /u/emaleau | 3607 | (80 posts)
3 | /u/shercocked | 2465 | (34 posts)
4 | /u/ruthannr94 | 2386 | (56 posts)
5 | /u/e___e | 2183 | (38 posts)
6 | /u/kcwashere | 1991 | (41 posts)
7 | /u/BlueSpiritBlues | 1904 | (27 posts)
8 | /u/sarahaw13 | 1785 | (42 posts)
9 | /u/full_boyle | 1726 | (18 posts)
10 | /u/villonious | 1707 | (35 posts)
11 | /u/xomandark | 1703 | (49 posts)
12 | /u/niftynonsense | 1625 | (24 posts)
13 | /u/Deiji- | 1596 | (39 posts)
14 | /u/ayiyi | 1448 | (13 posts)
15 | /u/recycledstardust | 1411 | (18 posts)
16 | /u/useafuckingcoaster | 1373 | (32 posts)
17 | /u/exmechanistic | 1349 | (37 posts)
18 | /u/bikinikills | 1315 | (59 posts)
19 | /u/LisaXavier | 1312 | (42 posts)
20 | /u/thecurvynerd | 1230 | (31 posts)
21 | /u/somethings_nothings | 1203 | (18 posts)
22 | /u/hopia89 | 1170 | (24 posts)
23 | /u/heyyogagirl | 1107 | (44 posts)


rank | user | total points | number of posts
---- | ---- | ------------ | ---------------
1 | /u/coffee_for_dinner | 3628 | (74 posts)
2 | /u/emaleau | 3607 | (80 posts)
3 | /u/shercocked | 2465 | (34 posts)
4 | /u/ruthannr94 | 2386 | (56 posts)
5 | /u/e___e | 2183 | (38 posts)
6 | /u/kcwashere | 1991 | (41 posts)
7 | /u/BlueSpiritBlues | 1904 | (27 posts)
8 | /u/sarahaw13 | 1785 | (42 posts)
9 | /u/full_boyle | 1726 | (18 posts)
10 | /u/villonious | 1707 | (35 posts)
11 | /u/xomandark | 1703 | (49 posts)
12 | /u/niftynonsense | 1625 | (24 posts)
13 | /u/Deiji- | 1596 | (39 posts)
14 | /u/ayiyi | 1448 | (13 posts)
15 | /u/recycledstardust | 1411 | (18 posts)
16 | /u/useafuckingcoaster | 1373 | (32 posts)
17 | /u/exmechanistic | 1349 | (37 posts)
18 | /u/bikinikills | 1315 | (59 posts)
19 | /u/LisaXavier | 1312 | (42 posts)
20 | /u/thecurvynerd | 1230 | (31 posts)
21 | /u/somethings_nothings | 1203 | (18 posts)
22 | /u/hopia89 | 1170 | (24 posts)
23 | /u/heyyogagirl | 1107 | (44 posts)
24 | /u/Lillyeth | 1079 | (25 posts)
25 | /u/respectablerag | 1059 | (38 posts)
26 | /u/lizzybones | 1032 | (28 posts)
27 | /u/saphydoodle | 982 | (10 posts)
28 | /u/Daniellewhatever | 978 | (39 posts)
29 | /u/foxlie | 977 | (20 posts)
30 | /u/0001909 | 965 | (10 posts)
31 | /u/bean-sprout | 939 | (14 posts)
32 | /u/ModestMalka | 891 | (20 posts)
33 | /u/BaboonStick | 886 | (16 posts)
34 | /u/muzzyhoo92 | 881 | (12 posts)
35 | /u/stephamelon | 848 | (18 posts)
36 | /u/Gabwielle | 842 | (18 posts)
37 | /u/ratioyouknow | 812 | (25 posts)
38 | /u/LizzyLemonade | 771 | (38 posts)
39 | /u/chaosintejas | 757 | (17 posts)
40 | /u/super_cheeky | 744 | (13 posts)
41 | /u/earth_rises | 718 | (16 posts)
42 | /u/salted_sky | 705 | (15 posts)
43 | /u/nipplemonger | 675 | (3 posts)
44 | /u/tomlizzo | 664 | (16 posts)
45 | /u/cheesegrater_knees | 640 | (21 posts)
46 | /u/ceruleanmuse | 639 | (10 posts)
47 | /u/xtinamariet | 618 | (42 posts)
48 | /u/SusannaCarmichael | 600 | (8 posts)
49 | /u/madinmadame | 595 | (9 posts)
50 | /u/babychickyellow | 594 | (8 posts)
51 | /u/charts_and_facts | 585 | (11 posts)
52 | /u/Hoosiergirl29 | 583 | (19 posts)
53 | /u/sohungy | 581 | (7 posts)
54 | /u/fitzpleasr | 561 | (13 posts)
55 | /u/Wasmand | 559 | (5 posts)
56 | /u/layneepup | 555 | (14 posts)
57 | /u/seriicis | 543 | (14 posts)
58 | /u/catherine_wheel | 538 | (15 posts)
59 | /u/faux-kwah | 531 | (9 posts)
60 | /u/eyeslikeashowroom | 523 | (15 posts)
61 | /u/tamberra | 521 | (16 posts)
62 | /u/Straightgrizzly22 | 520 | (7 posts)
63 | /u/goldiehawn | 515 | (6 posts)
64 | /u/pygoscelis | 489 | (18 posts)
65 | /u/charity1021 | 478 | (3 posts)
66 | /u/gombrick | 466 | (8 posts)
67 | /u/2000srh | 464 | (12 posts)
68 | /u/robotgwen | 464 | (24 posts)
69 | /u/sheena2015 | 443 | (15 posts)
70 | /u/wanderedoff | 440 | (9 posts)
71 | /u/KodiakTheBear9 | 437 | (10 posts)
72 | /u/corinnaps | 433 | (7 posts)
73 | /u/302149 | 433 | (24 posts)
74 | /u/ottaline_gambol | 419 | (12 posts)
75 | /u/tuliphaze | 416 | (11 posts)
76 | /u/Fiberista | 405 | (9 posts)
77 | /u/ticklytiger | 400 | (16 posts)
78 | /u/asdfjklOHFUCKYOU | 387 | (24 posts)
79 | /u/Scrap-Icon | 382 | (8 posts)
80 | /u/firstthingmonday | 374 | (8 posts)
81 | /u/miss-nebula | 371 | (6 posts)
82 | /u/cinnathymee | 368 | (15 posts)
83 | /u/caffeinatedlackey | 368 | (13 posts)
84 | /u/kinasekinase | 361 | (15 posts)
85 | /u/ediblesprysky | 355 | (18 posts)
86 | /u/-beyonce- | 351 | (9 posts)
87 | /u/w0nd3rlust | 350 | (14 posts)
88 | /u/KimchiChan | 349 | (9 posts)
89 | /u/PM_ME_ITTYBITTYKITTY | 335 | (4 posts)
90 | /u/undercoverofbranches | 329 | (33 posts)
91 | /u/carmen0042 | 326 | (4 posts)
92 | /u/imogenimpulse | 316 | (6 posts)
93 | /u/always-_-sleepy | 313 | (6 posts)
94 | /u/darkitecture | 313 | (6 posts)
95 | /u/Rogue_Fibre | 310 | (13 posts)
96 | /u/lawschoolbimbo | 302 | (15 posts)
97 | /u/hiilive | 298 | (5 posts)
98 | /u/ahrieru | 298 | (5 posts)
99 | /u/suckckit | 288 | (6 posts)
100 | /u/MoonlightBloom | 288 | (7 posts)

In [114]:
for user in users:
    users[user]['points_per_post'] = users[user]['points'] / users[user]['count']

by_ratio = sorted(users, key=lambda x: users[x]['points_per_post'], reverse=True)
print('rank | user | points per post | number of posts')
print('---- | ---- | --------------- | ---------------')
for i,user in enumerate(by_ratio[:100]):
    print("{} | /u/{} | {} | ({} posts)".format(i+1, user, users[user]['points_per_post'], users[user]['count']))

rank | user | points per post | number of posts
---- | ---- | --------------- | ---------------
1 | /u/nipplemonger | 225.0 | (3 posts)
2 | /u/minka92 | 200.0 | (1 posts)
3 | /u/pandawww | 194.0 | (1 posts)
4 | /u/maker_take_you | 170.0 | (1 posts)
5 | /u/charity1021 | 159.33333333333334 | (3 posts)
6 | /u/Cubone_bone | 154.0 | (1 posts)
7 | /u/CplElise | 151.0 | (1 posts)
8 | /u/nAVEEE | 150.0 | (1 posts)
9 | /u/thebigmeowski | 141.0 | (1 posts)
10 | /u/missgracie | 137.0 | (1 posts)
11 | /u/generalusername35 | 127.0 | (1 posts)
12 | /u/pollydog | 125.0 | (1 posts)
13 | /u/Metcarfre | 117.0 | (1 posts)
14 | /u/zoxe | 116.0 | (1 posts)
15 | /u/sewing_magic | 114.0 | (2 posts)
16 | /u/Wasmand | 111.8 | (5 posts)
17 | /u/ayiyi | 111.38461538461539 | (13 posts)
18 | /u/gnarlykitten | 105.0 | (1 posts)
19 | /u/DrNooooooo | 104.0 | (1 posts)
20 | /u/pilnock | 100.0 | (1 posts)
21 | /u/xiaojyao | 100.0 | (2 posts)
22 | /u/skyllanyx | 99.5 | (2 posts)
23 | /u/saphydoodle | 98.2 | (10 posts)
2

rank | user | points per post | number of posts
---- | ---- | --------------- | ---------------
1 | /u/nipplemonger | 225.0 | (3 posts)
2 | /u/minka92 | 200.0 | (1 posts)
3 | /u/pandawww | 194.0 | (1 posts)
4 | /u/maker_take_you | 170.0 | (1 posts)
5 | /u/charity1021 | 159.33333333333334 | (3 posts)
6 | /u/Cubone_bone | 154.0 | (1 posts)
7 | /u/CplElise | 151.0 | (1 posts)
8 | /u/nAVEEE | 150.0 | (1 posts)
9 | /u/thebigmeowski | 141.0 | (1 posts)
10 | /u/missgracie | 137.0 | (1 posts)
11 | /u/generalusername35 | 127.0 | (1 posts)
12 | /u/pollydog | 125.0 | (1 posts)
13 | /u/Metcarfre | 117.0 | (1 posts)
14 | /u/zoxe | 116.0 | (1 posts)
15 | /u/sewing_magic | 114.0 | (2 posts)
16 | /u/Wasmand | 111.8 | (5 posts)
17 | /u/ayiyi | 111.38461538461539 | (13 posts)
18 | /u/gnarlykitten | 105.0 | (1 posts)
19 | /u/DrNooooooo | 104.0 | (1 posts)
20 | /u/pilnock | 100.0 | (1 posts)
21 | /u/xiaojyao | 100.0 | (2 posts)
22 | /u/skyllanyx | 99.5 | (2 posts)
23 | /u/saphydoodle | 98.2 | (10 posts)
24 | /u/orlando_furiosa | 97.0 | (1 posts)
25 | /u/0001909 | 96.5 | (10 posts)
26 | /u/nachtmere | 96.0 | (1 posts)
27 | /u/full_boyle | 95.88888888888889 | (18 posts)
28 | /u/LentilBurguer | 95.0 | (1 posts)
29 | /u/Bloopity_bloop_bloop | 94.0 | (1 posts)
30 | /u/nnutella | 91.0 | (1 posts)
31 | /u/theteenagegentleman | 90.0 | (1 posts)
32 | /u/cadmiumred | 87.0 | (2 posts)
33 | /u/goldiehawn | 85.83333333333333 | (6 posts)
34 | /u/PM_ME_ITTYBITTYKITTY | 83.75 | (4 posts)
35 | /u/sohungy | 83.0 | (7 posts)
36 | /u/social_potato | 83.0 | (1 posts)
37 | /u/carmen0042 | 81.5 | (4 posts)
38 | /u/chameshi_nampa | 81.0 | (1 posts)
39 | /u/maya1987_K | 81.0 | (3 posts)
40 | /u/cccoffeeegurl | 80.0 | (1 posts)
41 | /u/mga92 | 80.0 | (1 posts)
42 | /u/recycledstardust | 78.38888888888889 | (18 posts)
43 | /u/gal_gardner | 78.0 | (2 posts)
44 | /u/therealtompetty | 77.0 | (1 posts)
45 | /u/amber0517 | 76.0 | (1 posts)
46 | /u/viabrera | 76.0 | (1 posts)
47 | /u/SusannaCarmichael | 75.0 | (8 posts)
48 | /u/any_delirium | 75.0 | (1 posts)
49 | /u/smolherring | 74.33333333333333 | (3 posts)
50 | /u/Straightgrizzly22 | 74.28571428571429 | (7 posts)
51 | /u/babychickyellow | 74.25 | (8 posts)
52 | /u/Berryjammy | 73.5 | (2 posts)
53 | /u/muzzyhoo92 | 73.41666666666667 | (12 posts)
54 | /u/shercocked | 72.5 | (34 posts)
55 | /u/filletoxico | 72.0 | (2 posts)
56 | /u/ilovemrmiyagi | 72.0 | (1 posts)
57 | /u/rch-ie | 71.0 | (1 posts)
58 | /u/BlueSpiritBlues | 70.51851851851852 | (27 posts)
59 | /u/sudosussudio | 70.0 | (1 posts)
60 | /u/Sun_Sprout | 70.0 | (1 posts)
61 | /u/feminines | 69.5 | (2 posts)
62 | /u/clankton | 69.0 | (1 posts)
63 | /u/mycatispipeyq | 68.0 | (4 posts)
64 | /u/_ad_nauseam | 68.0 | (1 posts)
65 | /u/niftynonsense | 67.70833333333333 | (24 posts)
66 | /u/bean-sprout | 67.07142857142857 | (14 posts)
67 | /u/AVOCADOCUPCAKES | 67.0 | (4 posts)
68 | /u/somethings_nothings | 66.83333333333333 | (18 posts)
69 | /u/ohmamie | 66.5 | (2 posts)
70 | /u/madinmadame | 66.11111111111111 | (9 posts)
71 | /u/Olibrow14 | 66.0 | (1 posts)
72 | /u/casevintageshop | 66.0 | (1 posts)
73 | /u/thane311 | 65.0 | (1 posts)
74 | /u/jjcreskoff | 65.0 | (1 posts)
75 | /u/averagefruit | 64.0 | (1 posts)
76 | /u/ceruleanmuse | 63.9 | (10 posts)
77 | /u/WesterosiAssassin | 63.0 | (1 posts)
78 | /u/Kkal73 | 62.5 | (2 posts)
79 | /u/b_writes | 62.5 | (4 posts)
80 | /u/maydaytuesday | 62.0 | (1 posts)
81 | /u/corinnaps | 61.857142857142854 | (7 posts)
82 | /u/miss-nebula | 61.833333333333336 | (6 posts)
83 | /u/yesnomaybe-idontknow | 61.0 | (1 posts)
84 | /u/DigitalClutter | 60.0 | (1 posts)
85 | /u/hiilive | 59.6 | (5 posts)
86 | /u/ahrieru | 59.6 | (5 posts)
87 | /u/hand2handwombat | 59.5 | (4 posts)
88 | /u/faux-kwah | 59.0 | (9 posts)
89 | /u/gombrick | 58.25 | (8 posts)
90 | /u/691337 | 58.0 | (1 posts)
91 | /u/evrtgurnot | 58.0 | (2 posts)
92 | /u/e___e | 57.44736842105263 | (38 posts)
93 | /u/super_cheeky | 57.23076923076923 | (13 posts)
94 | /u/warpweftwatergate | 57.0 | (1 posts)
95 | /u/pamperedpunk | 57.0 | (1 posts)
96 | /u/oy_ | 56.666666666666664 | (3 posts)
97 | /u/papercranium | 56.5 | (2 posts)
98 | /u/EpicShitTime | 56.0 | (1 posts)
99 | /u/TopGagne | 56.0 | (1 posts)
100 | /u/BaboonStick | 55.375 | (16 posts)

In [115]:
users_set = set(users.keys())
len(users_set)

974

In [116]:
total_upvotes = sum([users[user]['points'] for user in users])
print(total_upvotes)

132641


In [117]:
total_posts = sum([users[user]['count'] for user in users])
print(total_posts)

3898


In [118]:
participated_once = sum([1 if users[user]['count']==1 else 0 for user in users])
print(participated_once)

482
